packages/zowe-explorer/src/uss/ZoweUSSNode.ts

```diff
@@ -630,23 +651,35 @@ export class ZoweUSSNode extends ZoweTreeNode implements IZoweUSSTreeNode {
         };
         try {
             const api = ZoweExplorerApiRegister.getUssApi(this.profile);
-            const apiResponse = await api.fileList(remotePath);
-            const fileList = apiResponse.apiResponse?.items;
-
-            for (const localFile of localFileNames) {
-                if (localFile.endsWith("/")) {
-                    const directoryPath = localFile.slice(0, -1);
-                    await api.uploadDirectory(directoryPath, remotePath + "/" + path.basename(directoryPath), options);
-                } else {
-                    let fname = path.basename(localFile);
-                    if (
-                        fileList?.find((file) => {
-                            return file.name === fname;
-                        })
-                    ) {
-                        fname += "(copy)";
+            const fileTreeToPaste: UssFileTree = JSON.parse(await vscode.env.clipboard.readText());
+
+            if (api.copy && UssFileUtils.toSameSession(fileTreeToPaste, this.getSessionNode().getLabel() as string)) {
+                for (const subnode of fileTreeToPaste.children) {
+                    await this.pasteFileTree(remotePath, subnode, {
```

Comment:
I like how we are checking for `api.copy` before we call this method.
However, there is nothing preventing someone from calling this public function in the future.
If it's not too much trouble, I will suggest adding some JSDoc to the `pasteFileTree` OR better yet...
Integrating the `if(api.copy)` close to the actual call (within this method).

Reply:
When making changes I realized why I kept the `api.copy` check separated from `pasteFileTree` - we only want to use this logic if the user is copying/pasting within the same LPAR, but only if `api.copy` is available.

So, I took both considerations into account - I added comments to the `pasteFileTree` function and a check for `copy/fileList` inside, in case users want to call this function for other reasons.

I'm no longer passing `options` to `pasteFileTree` as it this endpoint is not set up to support the same options within the CLI. I can always update this in the future once those options are accounted for within the `zowe.Utilities.putUSSPayload` function.